compiler/rustc_target/src/asm/avr.rs

```diff
@@ -106,6 +106,7 @@ def_regs! {
             "the stack pointer cannot be used as an operand for inline asm",
         #error = ["r0", "r1", "r1r0"] =>
             "r0 and r1 are not available due to an issue in LLVM",
+            // if this issue is resolved, then R0 has to be added to clobbered registers
```

Comment:
We need to carefully keep track of LLVM here: if LLVM ceases to treat these registers as reserved then we must immediately add them to the set of clobbered registers, otherwise incorrect code will be generated. This comment should be updated to reflect this.

Also, while reviewing the GCC docs, it seems that R1 always holds a value of 0. If this is a guarantee that asm can use then it should be documented, similar to how on x86 we guarantee the direction flag is always clear when entering or exiting inline asm.

Reply:
It's enough to just reword the comment.

Reply:
I've reworded the comment, please have a look if this works for you.

Regarding the R1 register content: the unstable-book already documents, that the value needs to be restored if altered:
https://github.com/rust-lang/rust/blob/17a19e684cdf3ca088af8b4da6a6209d128913f4/src/doc/unstable-book/src/language-features/asm-experimental-arch.md?plain=1#L156